service/src/test/java/com/ge/predix/controller/test/SubjectPrivilegeManagementControllerIT.java

```diff
@@ -82,16 +85,34 @@
     private ConfigurableEnvironment env;
 
     @BeforeClass
-    public void setup() throws Exception {
+    public void setup() {
         this.testZone = TEST_UTILS.createTestZone("SubjectMgmtControllerIT");
         this.testZone2 = TEST_UTILS.createTestZone("SubjectMgmtControllerIT2");
         this.zoneService.upsertZone(this.testZone);
         this.zoneService.upsertZone(this.testZone2);
+    }
+
+    @BeforeMethod
+    public void beforeMethod() {
         MockSecurityContext.mockSecurityContext(this.testZone);
         MockAcsRequestContext.mockAcsRequestContext(this.testZone);
     }
 
     @Test
+    public void subjectZoneDoesNotExistException() throws Exception {
+        // NOTE: To throw a ZoneDoesNotExistException, we must ensure that the AcsRequestContext in the
+        //       SpringSecurityZoneResolver class returns a null ZoneEntity
+        MockSecurityContext.mockSecurityContext(null);
+        MockAcsRequestContext.mockAcsRequestContext(this.testZone);
+
+        BaseSubject subject = JSON_UTILS.deserializeFromFile("controller-test/a-subject.json", BaseSubject.class);
+        MockMvcContext putContext = TEST_UTILS.createWACWithCustomPUTRequestBuilder(this.wac,
+                "zoneDoesNotExist", SUBJECT_BASE_URL + '/' + subject.getSubjectIdentifier());
+        putContext.getMockMvc().perform(putContext.getBuilder().contentType(MediaType.APPLICATION_JSON).
+                content(OBJECT_MAPPER.writeValueAsString(subject))).andExpect(status().isBadRequest());
```

Comment:
Mock ZoneDoesNotExistException throw to make sure this is what caused bad request status.